src/stream.js

```diff
@@ -0,0 +1,264 @@
+import { options, Fragment } from 'preact';
+import stream from 'stream';
+
+// components without names, kept as a hash for later comparison to return consistent UnnamedComponentXX names.
+const UNNAMED = [];
+
+const VOID_ELEMENTS = /^(area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)$/;
+
+
+function PreactReadableStream(vnode, context, opts) {
+	if (opts && opts.pretty) {
+		throw new Error('pretty is not supported in renderToNodeStream!');
+	}
+
+	stream.Readable.call(this, opts && opts.readable);
+
+	this.vnode = vnode;
+	this.context = context || {};
+	this.opts = opts || {};
+}
+
+PreactReadableStream.prototype = new stream.Readable();
+PreactReadableStream.prototype.constructor = PreactReadableStream;
+PreactReadableStream.prototype._read = function _read() {
+	try {
+		if (!this._generator) {
+			this._generator = this._generate(this.vnode, this.context, this.opts);
+		}
+		else if (this.reading) {
+			console.warn(new Error('You should not call PreactReadableStream#_read when a read is in progress').stack);
+		}
+
+		this.reading = true;
+
+		for (const chunk of this._generator) {
+			if (!this.push(chunk)) {
+				this.reading = false;
+				// high water mark reached, pause the stream until _read is called again...
+				return;
+			}
+		}
+	}
+	catch (e) {
+		this.emit('error', e);
+		this.push(null);
+		return;
+	}
+
+	// end the stream
+	this.push(null);
+};
+
+PreactReadableStream.prototype._generate = function *_generate(vnode, context, opts, inner, isSvgMode, selectValue) {
+	if (vnode==null || typeof vnode==='boolean') {
+		yield '';
+		return;
+	}
+
+	let nodeName = vnode.type,
+		props = vnode.props,
+		isComponent = false;
+	context = context || {};
+	opts = opts || {};
+
+	// #text nodes
+	if (typeof vnode!=='object' && !nodeName) {
+		yield encodeEntities(vnode);
+		return;
+	}
+
+	// components
+	if (typeof nodeName==='function') {
+		isComponent = true;
+		if (opts.shallow && (inner || opts.renderRootComponent===false)) {
+			nodeName = getComponentName(nodeName);
+		}
+		else if (nodeName===Fragment) {
+			let children = [];
+			getChildren(children, vnode.props.children);
+
+			for (let i = 0; i < children.length; i++) {
+				for (const chunk of this._generate(children[i], context, opts, opts.shallowHighOrder!==false, isSvgMode, selectValue)) {
+					yield chunk;
+				}
+			}
+
+			return;
+		}
+		else {
+			let c = vnode.__c = { __v: vnode, context, props: vnode.props };
+			if (options.render) options.render(vnode);
+
+			let renderedVNode;
+
+			if (!nodeName.prototype || typeof nodeName.prototype.render!=='function') {
+				// Necessary for createContext api. Setting this property will pass
+				// the context value as `this.context` just for this component.
+				let cxType = nodeName.contextType;
+				let provider = cxType && context[cxType.__c];
+				let cctx = cxType != null ? (provider ? provider.props.value : cxType._defaultValue) : context;
```

Comment:
Hello! Looks like the use of `_defaultValue` might be a bug (#130); changing it to `__` should fix it.